symfit/core/objectives.py

```diff
@@ -69,7 +69,34 @@ def __call__(self, ordered_parameters=[], **parameters):
         # zip will stop when the shortest of the two is exhausted
         parameters.update(dict(zip(self.model.free_params, ordered_parameters)))
         parameters.update(self.invariant_kwargs)
-        return self.model(**key2str(parameters))
+        result = self.model(**key2str(parameters))
+        return self._shape_dependent_data(result)
+
+    def _shape_dependent_data(self, model_output, param_level=0):
+        """
+        In rare cases, the dependent data and the output of the model do not
+        have the same shape. Think for example about :math:`y_i = a`. This
+        function makes sure both sides of the equation have the same shape.
+
+        :param model_output: Output of a call to model
+        :param param_level: indicates how many parameter dimensions should be
+            added to the shape. 0 for __call__, 1 for jac, 2 for hess.
+        :return: ``model_output`` reshaped to ``dependent_data``'s shape, if
+            possible.
+        """
+        shaped_result = []
+        n_params = len(self.model.params)
+        for dep_data, component in zip(self.dependent_data.values(), model_output):
```

Comment:
Are these guaranteed to have the same order on all versions of Python?

Reply:
Yes, I use always use protection: everything is an OrderedDict.